jabgui/src/main/java/org/jabref/gui/frame/JabRefFrameViewModel.java

```diff
@@ -165,16 +162,20 @@ public void handleUiCommands(List<UiCommand> uiCommands) {
         // Handle OpenDatabases
         if (!blank) {
             uiCommands.stream()
-                    .filter(UiCommand.OpenDatabases.class::isInstance)
-                    .map(UiCommand.OpenDatabases.class::cast)
-                    .forEach(command -> openDatabases(command.parserResults()));
-        }
+                    .filter(UiCommand.OpenLibraries.class::isInstance)
+                    .map(UiCommand.OpenLibraries.class::cast)
+                    .forEach(command -> openDatabaseAction.get().openFiles(command.toImport()));
 
-        // Handle automatically setting file links
-        uiCommands.stream()
-                  .filter(UiCommand.AutoSetFileLinks.class::isInstance).findAny()
-                  .map(UiCommand.AutoSetFileLinks.class::cast)
-                  .ifPresent(autoSetFileLinks -> autoSetFileLinks(autoSetFileLinks.parserResults()));
+            uiCommands.stream()
+                      .filter(UiCommand.AppendToCurrentLibrary.class::isInstance)
+                      .map(UiCommand.AppendToCurrentLibrary.class::cast)
+                      .map(UiCommand.AppendToCurrentLibrary::toAppend)
```

Comment:
A micro-optimization could be reducing an extra pass through stream? (although I like this syntax better^)
```suggestion
                      .map(command -> ((UiCommand.AppendToCurrentLibrary) command).toAppend())
```

Reply:
Semantically exactly the same. I don't think that this has any effect on bytecode.

Reply:
This is interesting to geek on, I'll try with some test using the eclipse jdk23 we use and get back

Reply:
So I converted this class to minimal number-based examples, using AI, and analyzed the bytecode diff (again, using AI) - here is what I got:
>1. Code Size:
TwoMapsApproach.class: 2203 bytes
OneMapApproach.class: 1974 bytes
The two-map approach generates more bytecode (approximately 11.6% larger)

>2. Method Implementation:
TwoMapsApproach: Uses method references with 4 bootstrap methods
OneMapApproach: Uses lambdas with 3 bootstrap methods

>3. Stack Operations (in processCommand method):
TwoMapsApproach: stack=3, locals=1
OneMapApproach: stack=2, locals=1
The two-map approach needs a larger operand stack

>4. Additional Operations in TwoMapsApproach:
Extra requireNonNull calls (lines 7 and 24)
Extra dup operations (lines 6 and 23)
Extra pop operations (lines 10 and 27)

>5. Method References vs Lambdas:
TwoMapsApproach uses direct method references (REF_invokeVirtual) for isInstance and cast
OneMapApproach uses synthetic lambda methods (lambda$processCommand$0, lambda$processCommand$1)

I understood that the bytecode optimization isn't the same, but didn't understand the jargon, so ran a [performance test](https://github.com/subhramit/map-test/blob/master/src/MapPerformanceTest.java). I've stored some results [here](https://github.com/subhramit/map-test/blob/master/log.txt). With with 10000000 elements, one map was 20-27% faster. For 50000000 elements, it was around 25-26% faster, but that was including two additional filter operations and a count:
```java
long count = java.util.Arrays.stream(commands)
                                     .filter(AppendToCurrentLibrary.class::isInstance)
                                     .map(AppendToCurrentLibrary.class::cast)
                                     .map(AppendToCurrentLibrary::toAppend)
                                     .filter(java.util.Objects::nonNull)
                                     .count();
```
vs
```java
long count = java.util.Arrays.stream(commands)
                                     .filter(command -> command instanceof AppendToCurrentLibrary)
                                     .map(command -> ((AppendToCurrentLibrary) command).toAppend())
                                     .filter(java.util.Objects::nonNull)
                                     .count();
```

Reply:
Given the fact, that uiCommands is a List consiting only of about max 4 to 6 Elements created on startup and AppendToCurrentLibrary is in 99.9999% of the cases used singularily from GUI-Cli, this is probably negligible, so I opt for better readability... :-D

Reply:
Or in other words: You got too much time. 😄

Reply:
Interesting... My bet would have been that the compiler optimizes that...

OMG... Please keep this noted - for some Java blog articles - or JDK input :)